### PR TITLE
Disable the international labels feature if the plugin isn't at least a specified version

### PR DESCRIPTION
**Pull request by DanReyLop**
The international shipping labels feature has substantial changes in the PHP plugin side, so we've decided to disable it entirely in Calypso if the plugin doesn't have the required changes.

There are 2 ways of doing this:
* The same way we've been dealing with backwards-compatibility up to now: Check an endpoint response for a particular flag / property. If it has it, then we can be sure that the plugin has the most up-to-date code. If it doesn't, it means the plugin code is from before that flag/property was introduced. This approach is favored by (at least) @marcinbot because it makes it easier to test locally.
* Enable/disable the feature by plugin version. This has the downside of having to manually edit the `Version` header comment of the plugin to test this feature before a compatible version of the plugin is released.

I took the "disable by version" approach because it's simpler and faster. If the plugin is less than `1.16.0`, we don't even need to fetch anything from the REST API, we can just disable the feature.

Are you folks OK with this approach? Honestly, I had this code sitting in my computer and I don't want to try the other approach without at least checking if this one is good enough.

What could we do to make this easier to live with? Maybe ignore the version check in `development`? Or bump the plugin version to `1.16.0` in the plugin's `add/intl-labels` branch? I think this approach would make life easier for us in the future because we could just gate features to newer versions of the plugin, without bending ourselves backwards to handle backwards-compatibility.

To test:
* Go to an international order (origin in the US, destination outside the US).
* Check that the `Print label` flow isn't available, instead the simple `Fulfill` action button is displayed.
* Go to your site, change the version in the header of the `woocommerce-services/woocommerce-services.php` file to `1.16.0` (or bigger).
* Refresh the Calypso page.
* Check that you can print a label now.

### CODE REVIEW

File: client/extensions/woocommerce/state/selectors/plugins.js
Comment (jeffstieler) [inline client/extensions/woocommerce/state/selectors/plugins.js]: Rather than define a new selector, can we just use [`getPluginOnSite()`](https://github.com/Automattic/wp-calypso/blob/master/client/state/plugins/installed/selectors.js#L97)?

**Review by jeffstieler — APPROVED** [review]
Code looks good (with one comment about reusing a selector) and tests well. 👍 

Approving for merge pending the selector change.

File: client/extensions/woocommerce/state/selectors/plugins.js
Comment (DanReyLop) [reply client/extensions/woocommerce/state/selectors/plugins.js]: The primary reason is that I didn't know it existed. I found a couple of things:
* `getPluginOnSite()` isn't memoized and it does a non-trivial amount of work.
* `getPluginOnSite()` can't filter by `active`, and the returned data structure is something like `{ version: '1.2.3', sites: { 357532: { active: true } } }`

I've refactored the new selector to use `getPluginOnSite()` but I kept it to keep the memoization and a consistent interface (returns `null` if the plugin is not active). Is that better?

**Comment by DanReyLop** [comment]
> eslint failing now on this one

This PR is going to be merged into the `add/store-intl-labels-ui` branch, which has the same `eslint` file failing, so it's no problem.

**Comment by DanReyLop** [comment]
I fixed the test that was failing. The `eslint` error is already happening in the target branch (`add/store-intl-labels-ui`) so it doesn't matter for this PR.

**Comment by jeffstieler** [comment]
@alisterscott I don't think we should block on the failing `eslint` test since this is destined for `add/store-intl-labels-ui` not `master`. That OK?

**Comment by jeffstieler** [comment]
We will fix the `eslint` error before a merge to `master` - making the call to merge this now.

**Pull request merged**
